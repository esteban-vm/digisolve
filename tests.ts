import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
