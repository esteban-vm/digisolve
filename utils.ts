export const formatPrice = (price: number) => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 0,
  }).format(price)
}
