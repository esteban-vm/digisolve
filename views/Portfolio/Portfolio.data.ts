import type { WorkProps } from './Work'

const width = 394
const height = 470

const works: WorkProps[] = [
  {
    id: '0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a',
    text: 'Online advertising',
    image: { src: '/img/online-advertising.png', width, height },
  },
  {
    id: '35501350-65b2-47bc-95a8-97bc27d2e186',
    text: 'Social media campaigns',
    image: { src: '/img/social-media-campaigns.png', width, height },
  },
  {
    id: '26c86632-8302-4ba4-b316-0a810712efef',
    text: 'UX design',
    image: { src: '/img/ux-design.png', width, height },
  },
  {
    id: 'e333c170-6d2a-43e9-9720-0a1d50d9c3ba',
    text: 'Branding & digital marketing',
    image: { src: '/img/branding-digital-marketing.png', width, height },
  },
]

export default works
