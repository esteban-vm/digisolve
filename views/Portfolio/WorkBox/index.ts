export { default, type WorkProps } from './WorkBox'
