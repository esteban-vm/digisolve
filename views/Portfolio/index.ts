export { default as Portfolio, works } from './Portfolio'
