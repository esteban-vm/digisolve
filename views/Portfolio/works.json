[
  {
    "title": "Online advertising",
    "image": {
      "src": "/img/online-advertising.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Social media campaigns",
    "image": {
      "src": "/img/social-media-campaigns.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "UX design",
    "image": {
      "src": "/img/ux-design.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Branding & digital marketing",
    "image": {
      "src": "/img/branding-digital-marketing.png",
      "height": 470,
      "width": 394
    }
  }
]
