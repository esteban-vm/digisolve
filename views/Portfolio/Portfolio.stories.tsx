import type { Meta } from '@storybook/react'
import Portfolio from './Portfolio'

export default {
  component: Portfolio,
  title: 'Views/Portfolio',
} as Meta<typeof Portfolio>

export const PortfolioView = () => <Portfolio />
