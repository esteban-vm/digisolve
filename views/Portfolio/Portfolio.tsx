import type { Component } from '@/types'
import { Grid } from '@/components'
import { uuid } from '@/helpers'
import { styled, mediaQuery } from '@/styles'
import works from './works.json'
import Work from './Work'

const PortfolioComponent: Component = (props) => {
  return (
    <section id='portfolio' aria-labelledby='portfolio_title' {...props}>
      <Grid.Row>
        <h2 id='portfolio_title' className='section__heading'>
          Our Work
        </h2>
        <h3 className='section__subheading'>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
        </h3>
      </Grid.Row>
      <ul className='clear_fix'>
        {works.map((work) => (
          <li key={uuid()}>
            <Work {...work} />
          </li>
        ))}
      </ul>
    </section>
  )
}

const Portfolio = styled(PortfolioComponent)`
  text-align: center;
  background: var(--color-light);

  ul {
    margin-top: 10px;
    width: 100%;

    li {
      display: block;
      width: 25%;
      float: left;

      ${mediaQuery('xs')} {
        width: 50%;
      }
    }
  }
`

export default Portfolio
