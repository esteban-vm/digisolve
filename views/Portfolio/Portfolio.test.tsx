import { render, cleanup, screen, create } from '@/utils/tests'
import Portfolio, { works } from './Portfolio'

describe('ðŸ§ª <Portfolio /> test cases:', () => {
  describe('there should be:', () => {
    beforeEach(() => {
      render(<Portfolio />)
    })

    afterEach(cleanup)

    it('a heading', () => {
      const heading = screen.getByRole('heading', { name: /^our work$/i })
      expect(heading).toBeInTheDocument()
      expect(heading).toBeVisible()
    })

    it('a subheading', () => {
      const [, subheading] = screen.getAllByRole('heading')
      expect(subheading).toBeInTheDocument()
      expect(subheading).toBeVisible()
    })

    it(`${works.length} work items`, () => {
      const numberOfWorks = works.length
      const workImages = screen.getAllByRole('img')
      const workTexts = screen.getAllByRole('paragraph')

      expect(workImages).toHaveLength(numberOfWorks)
      expect(workTexts).toHaveLength(numberOfWorks)

      for (let index = 0; index < numberOfWorks; index++) {
        const workImage = workImages[index]
        const workTextElem = workTexts[index]
        const workTextStr = works[index].text

        expect(workImage).toBeInTheDocument()
        expect(workImage).toBeVisible()
        expect(workImage).toHaveAccessibleName(workTextStr)

        expect(workTextElem).toBeInTheDocument()
        expect(workTextElem).toBeVisible()
        expect(workTextElem).toHaveTextContent(workTextStr)
      }
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const portfolioTree = create(<Portfolio />).toJSON()
      expect(portfolioTree).toMatchSnapshot()
    })
  })
})
