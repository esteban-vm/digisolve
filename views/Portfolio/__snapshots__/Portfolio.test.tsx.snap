// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª PORTFOLIO: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
  background: var(--color-light);
}

.emotion-0 ul {
  margin-top: 10px;
  width: 100%;
}

.emotion-0 ul li {
  display: block;
  width: 25%;
  float: left;
}

@media only screen and (max-width: 480px) {
  .emotion-0 ul li {
    width: 50%;
  }
}

.emotion-1 {
  background-color: var(--color-black);
  overflow: hidden;
}

.emotion-1 img {
  width: 100%;
  height: auto;
  opacity: 0.7;
  -webkit-transform: scale(1.15);
  -moz-transform: scale(1.15);
  -ms-transform: scale(1.15);
  transform: scale(1.15);
  -webkit-transition: -webkit-transform 500ms,opacity 500ms;
  transition: transform 500ms,opacity 500ms;
}

.emotion-1 img:hover {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.emotion-1 p {
  color: var(--color-white);
  text-transform: capitalize;
}

<section
  aria-labelledby="portfolio_title"
  className="emotion-0"
  id="portfolio"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      className="section__heading"
      id="portfolio_title"
    >
      Our Work
    </h2>
    <h3
      className="section__subheading"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
    </h3>
  </div>
  <ul
    className="clear_fix"
  >
    <li>
      <figure
        aria-labelledby="work_image"
        className="emotion-1"
      >
        <img
          alt="Online advertising"
          data-nimg="1"
          decoding="async"
          height={470}
          id="work_image"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Online advertising
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="work_image"
        className="emotion-1"
      >
        <img
          alt="Social media campaigns"
          data-nimg="1"
          decoding="async"
          height={470}
          id="work_image"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Social media campaigns
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="work_image"
        className="emotion-1"
      >
        <img
          alt="UX design"
          data-nimg="1"
          decoding="async"
          height={470}
          id="work_image"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fux-design.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fux-design.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fux-design.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          UX design
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="work_image"
        className="emotion-1"
      >
        <img
          alt="Branding & digital marketing"
          data-nimg="1"
          decoding="async"
          height={470}
          id="work_image"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Branding & digital marketing
        </p>
      </figure>
    </li>
  </ul>
</section>
`;
